packages/gestalt/src/Icon/Icon.js

```diff
@@ -58,15 +58,11 @@ export default function Icon(props: Props) {
     [styles.iconBlock]: !inline,
   });
 
-  let path;
-  if (icon) {
-    path = icons[icon];
-  } else if (dangerouslySetSvgPath) {
+  const path =
+    (icon && icons[icon]) ||
     /* eslint-disable-next-line no-underscore-dangle */
-    path = dangerouslySetSvgPath.__path;
-  } else {
-    path = '';
```

Comment:
I chose to remove this because Flow would not allow use to reach this line in a test unless we add `$ExpectError` or the likes. More importantly though, the native `<path />` component is ok getting passed `undefined` for the value of `d` because itll function like any other undefined attribute so imo this is actually better than an empty string.